pycheribuild/qemu_utils.py

```diff
@@ -89,6 +89,21 @@ def __init__(self, xtarget: CrossCompileTarget, want_debugger=False):
             raise ValueError("Unknown target " + str(xtarget))
 
     def disk_image_args(self, image: Path, image_format: str) -> list:
+        # Probe the disk image format in case someone has overridden the default image path or format is unspecified
+        if not image.exists():
+            # Either we're pretending or we'll complain elsewhere.
+            if image_format is None:
+                image_format = "qcow2" if image.name.endswith(".qcow2") else "raw"
+        else:
+            with image.open('rb') as imgf:
+                magic = imgf.read(4)
+                is_qcow2 = magic == b'QFI\xfb'
+
+                if image_format is None:
+                    image_format = "qcow2" if is_qcow2 else "raw"
+                elif is_qcow2 and image_format != "qcow2":
+                    self.warning("Disk image looks like qcow2 but image_format is %s" % image_format)
```

Comment:
Oh, hm, no, this not being a `Project` means there isn't a `self.warning` either (oh, for static typing).  Suggestions, @arichardson?